akka-docs/src/main/paradox/java/cluster-sharding.md

```diff
@@ -421,3 +421,13 @@ of entities that are alive in each shard.
 
 The purpose of these messages is testing and monitoring, they are not provided to give access to
 directly sending messages to the individual entities.
+
+## Rolling upgrades
+
+When doing rolling upgrades special care must be taken to not change any of the following aspects of sharding:
+
+ * the `extractShardId` function so that it extracts a different shard id than before for a given message
```

Comment:
Yeah, I see. I wanted to make a difference between adding new types of messages to it and changing what id it returns from an existing message. I'll see if I can rephrase.